account_netting/tests/test_account_netting.py

```diff
@@ -1,77 +1,134 @@
 # -*- coding: utf-8 -*-
-# (c) 2015 Pedro M. Baeza
+# Copyright 2015 Pedro M. Baeza
+# Copyright 2017 Tecnativa - Vicent Cubells
 # License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
 
 import openerp.tests.common as common
-from openerp import workflow
 
 
-class TestAccountNetting(common.TransactionCase):
-
-    def setUp(self):
-        super(TestAccountNetting, self).setUp()
-        self.partner = self.env['res.partner'].create(
+class TestAccountNetting(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestAccountNetting, cls).setUpClass()
+        cls.partner = cls.env['res.partner'].create(
             {'supplier': True,
              'customer': True,
              'name': "Supplier/Customer"
              })
-        self.invoice_model = self.env['account.invoice']
-        self.account_receivable = self.env.ref('account.a_recv')
-        self.customer_invoice = self.invoice_model.create(
-            {'journal_id': self.env.ref('account.sales_journal').id,
-             'type': 'out_invoice',
-             'partner_id': self.partner.id,
-             'account_id': self.account_receivable.id,
-             'invoice_line': [(0, 0, {'name': 'Test',
-                                      'price_unit': 100.0})],
-             })
-        workflow.trg_validate(
-            self.uid, 'account.invoice', self.customer_invoice.id,
-            'invoice_open', self.cr)
-        customer_move = self.customer_invoice.move_id
-        self.move_line_1 = customer_move.line_id.filtered(
-            lambda x: x.account_id == self.account_receivable)
-        self.account_payable = self.env.ref('account.a_pay')
-        self.supplier_invoice = self.invoice_model.create(
-            {'journal_id': self.env.ref('account.expenses_journal').id,
-             'type': 'in_invoice',
-             'partner_id': self.partner.id,
-             'account_id': self.account_payable.id,
-             'invoice_line': [(0, 0, {'name': 'Test',
-                                      'price_unit': 1200.0})],
-             })
-        workflow.trg_validate(
-            self.uid, 'account.invoice', self.supplier_invoice.id,
-            'invoice_open', self.cr)
-        supplier_move = self.supplier_invoice.move_id
-        self.move_line_2 = supplier_move.line_id.filtered(
-            lambda x: x.account_id == self.account_payable)
+        res_users_account_manager = cls.env.ref(
+            'account.group_account_manager')
+        partner_manager = cls.env.ref('base.group_partner_manager')
+        cls.account_manager = cls.env['res.users'].with_context(
```

Comment:
Don't create a new user, but just add the corresponding group to your user (KISS principle)

Reply:
This hasn't been attended

Reply:
You are right